### ISSUE TITLE
Show meaningful information on the session page while starting

### ISSUE THREAD

**Issue by lorenzo-cavazzi** [SwissDataScienceCenter/renku-ui#1879]
## Motivation

Sessions take time to start, 1 minute (or more) is not uncommon.
Especially when clicking on the quickstart button, users may be confused since they need to wait for such a long time with no feedback apart from the bouncing dots. The status line on top saying `Started x seconds ago` adds to the confusion.

## Mock-ups / Design

We could improve the feedback by:
* Changing the status line to `Starting`.
* Adding better feedback in the main window.
    - A short message saying the sessions are starting and it may take a minute or two
    - A progress bar instead of the bounding dots
    - Some feedback from the `status.message` content. It currently mentions the pods with a non-ready state, maybe that should be changed a bit. The idea is to give the user the perception that something is going on in the background. E.G. showing `starting x of 5 components`, or come up with a short descriptive sentence based on what still needs to start.

## Progress

- [ ] API Design (sprint-??)
- [ ] UI Implementation

**Comment by rokroskar** [comment]
This is how jupyterhub implements it: https://github.com/jupyterhub/jupyterhub/blob/6be3160d7487bb362df7095f24454ef2814951fc/share/jupyterhub/templates/spawn_pending.html

**Comment by andre-code** [comment]
@rokroskar It is possible to include the list of pending containers to start, do you think that information is useful for the user or the message `1 of n containers is enough`. ?

![Screenshot 2022-06-27 at 11 40 03](https://user-images.githubusercontent.com/43388408/175912732-0ecd4065-2f8f-4a52-bf3b-cee9cfbc5347.png)

**Comment by rokroskar** [comment]
I'm not sure having the names displayed is necessarily useful. Do we have information about what is happening with the individual containers, e.g. image being pulled?

**Comment by andre-code** [comment]
The message comes directly from notebook-service. So @olevski  probably has more information about it.

**Comment by olevski** [comment]
> Do we have information about what is happening with the individual containers, e.g. image being pulled?

The notebook service can get that information from k8s if you want to display it. It is a bit more complicated to implement than just saying 1 out of n containers are ready but definitely doable.

What would the message look like if we should what each individual container is doing? Is that too much information?

I guess it will be a dictionary with container names as keys and statuses/additional info as values. Currently we simply send a status (one of Running, Starting, Stopping, Failed) and a message (i.e. containers with unready status: a, b, c). These are directly then used by the ui.

**Comment by andre-code** [comment]
If it is possible to know the steps in advance, it would be great to receive the status like this:
```
const status =
{
  state: "starting",
  details:
    [
      {
        step: "Setting up support containers",
        status: "done", // pending, error, done
      },
      {
        step: "Cloning and configuring the repository",
        status: "pending", // pending, error, done
      },
      {
        step: "Pulling main image",
        status: "pending", // pending, error, done
      },
      {
        step: "Starting server",
        status: "pending", // pending, error, done
      },
    ],
};
```
That way it would change the way of presenting the progress of start a session in to something like: 

![Screenshot 2022-06-27 at 14 04 18](https://user-images.githubusercontent.com/43388408/175936954-683db20d-45bc-4c46-a08b-e1540da97c38.png)

**Comment by rokroskar** [comment]
@gavin-k-lee I agree - but the expansion should show what @andre-code has in the example above, not the container names, imho

**Comment by andre-code** [comment]
thanks @olevski. All at the same end point as you proposed.

**Comment by ksanao** [comment]
I recommend showing a progress bar (with the information on the completed steps that you suggest) and removing the timer.

**Comment by ciyer** [comment]
Also, in the session starting display, we should show a message indicating that the user will be brought into an autosave branch if that is the case.



![Image](https://user-images.githubusercontent.com/1196411/191684030-ecc3bd01-501e-418d-9483-70345496dfaa.png)
